src/Common/IntegerMaskParser.h

```diff
@@ -0,0 +1,84 @@
+#pragma once
+
+#include <string>
+#include <optional>
+#include <regex>
+#include <vector>
+
+
+namespace seissol {
+class IntegerMaskParser {
```

Comment:
Separate code in .h and .cpp. No point for header-only here as the class does not use templates.